src/offchain/lockTx.ts

```diff
@@ -1,59 +1,50 @@
-import koios from "./koios";
-
-async function getLockTx( wallet: BrowserWallet ): Promise<Tx>
-{
-    // creates an address form the bech32 form
-    const myAddr = Address.fromString(
-        await wallet.getChangeAddress()
-    );
-
-    const txBuilder = await getTxBuilder();
-    const myUTxOs = (await wallet.getUtxos()).map( toPlutsUtxo );
-
-    if( myUTxOs.length === 0 )
-    {
-        throw new Error("have you requested founds from the faucet?")
-    }
-
-    const utxo = myUTxOs.find( u => u.resolved.value.lovelaces > 15_000_000 );
-
-    if( utxo === undefined )
-    {
-        throw "not enough ada";
-    }
+import getTxBuilder from "./getTxBuilder";
 
-    return txBuilder.buildSync({
-        inputs: [{ utxo }],
-        outputs: [
-            { // output holding the founds that we'll spend later
-                address: scriptTestnetAddr,
-                // 10M lovelaces === 10 ADA
-                value: Value.lovelaces( 10_000_000 ),
-                // remeber to include a datum
-                datum: new DataB(
-                    // remember we set the datum to be the public key hash?
-                    // we can extract it from the address as follows
-                    myAddr.paymentCreds.hash.toBuffer()
-                )
-            }
-        ],
-        // send everything left back to us
-        changeAddress: myAddr
-    });
+async function getLockTx(wallet: BrowserWallet, Blockfrost: BlockfrostPluts): Promise<Tx> {
+  // creates an address form the bech32 form
+  const myAddr = Address.fromString(
+    await wallet.getChangeAddress()
+  );
+
+  const txBuilder = await getTxBuilder(Blockfrost);
+  const myUTxOs = (await wallet.getUtxos()).map(toPlutsUtxo);
+
+  if (myUTxOs.length === 0) {
+    throw new Error("have you requested founds from the faucet?");
+  }
+
+  const utxo = myUTxOs.find(u => u.resolved.value.lovelaces > 15_000_000);
+
+  if (utxo === undefined) {
+    throw new Error("not enough ada");
+  }
+
+  return txBuilder.buildSync({
+    inputs: [{ utxo }],
+    outputs: [{ // output holding the founds that we'll spend later
+      address: scriptTestnetAddr,
+      // 10M lovelaces === 10 ADA
+      value: Value.lovelaces(10_000_000),
+      // remeber to include a datum
+      datum: new DataB(
+        // remember we set the datum to be the public key hash?
+        // we can extract it from the address as follows
+        myAddr.paymentCreds.hash.toBuffer()
+      )
+    }],
+    // send everything left back to us
+    changeAddress: myAddr
+  });
 }
 
-export async function lockTx( wallet: BrowserWallet): Promise<string>
-{
-    const unsingedTx = await getLockTx( wallet );
-
-    const txStr = await wallet.signTx(
-        unsingedTx.toCbor().toString()
-    );
-
-    return (await koios.tx.submit( Tx.fromCbor( txStr ) as any )).toString();
+export async function lockTx(wallet: BrowserWallet, projectId: string): Promise<string> {
+  const Blockfrost = new BlockfrostPluts({ projectId });
```

Comment:
@EehMauro , 
Isn't a new Blockfrost instance created here on lock, and a different one for unlock when done this way. Shouldn't we be sharing a Blockfrost obj created from the project-api-key throughout the app.

Reply:
Yes, it's a new instance. But I think it's not expensive because creating new instances doesn't require any API calls, and allows us to update the `projectId` in case it changes between calls